aristanetworks/ansible-cvp
# Fix(cv_device_v3): Improve error message when no devices are match Issue #1293 on ansible-avd

## Change Summary

Added better error handling in scenarios where `device_filter` is set to a string which does not match any devices.

## Related Issue(s)

Fixes https://github.com/aristanetworks/ansible-avd/issues/1293

## Component(s) name

`arista.cvp.device_v3`

## Proposed changes


## How to test

Simply add a string that won't match on any devices

```
- name: Configuration deployment with CVP
  hosts: AVD-CVP-1
  connection: local
  gather_facts: no
  tasks:
    - name: run CVP provisioning
      import_role:
         name: arista.avd.eos_config_deploy_cvp
      vars:
        container_root: 'TP_FABRIC'
        configlets_prefix: 'tp-avd'
        cv_collection: v3
        device_filter: 'xyz123'
        state: present
```

The error message now should be much clearer

```
TASK [arista.avd.eos_config_deploy_cvp : Configure devices on AVD-CVP-1] *******************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AssertionError: The device_filter did not match any devices. Please correct the value of the filter.
fatal: [AVD-CVP-1]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/avd/.ansible/tmp/ansible-local-33823oeum8q1m/ansible-tmp-1635471117.7400463-35328-218641083322507/AnsiballZ_cv_device_v3.py\", line 100, in <module>\n    _ansiballz_main()\n  File \"/home/avd/.ansible/tmp/ansible-local-33823oeum8q1m/ansible-tmp-1635471117.7400463-35328-218641083322507/AnsiballZ_cv_device_v3.py\", line 92, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/avd/.ansible/tmp/ansible-local-33823oeum8q1m/ansible-tmp-1635471117.7400463-35328-218641083322507/AnsiballZ_cv_device_v3.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.arista.cvp.plugins.modules.cv_device_v3', init_globals=dict(_module_fqn='ansible_collections.arista.cvp.plugins.modules.cv_device_v3', _modlib_path=modlib_path),\n  File \"/usr/local/lib/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/local/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/local/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_arista.cvp.cv_device_v3_payload_exge3orm/ansible_arista.cvp.cv_device_v3_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py\", line 217, in <module>\n  File \"/tmp/ansible_arista.cvp.cv_device_v3_payload_exge3orm/ansible_arista.cvp.cv_device_v3_payload.zip/ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py\", line 194, in main\n  File \"/tmp/ansible_arista.cvp.cv_device_v3_payload_exge3orm/ansible_arista.cvp.cv_device_v3_payload.zip/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py\", line 266, in __init__\nAssertionError: The device_filter did not match any devices. Please correct the value of the filter.\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

```

## Checklist

### User Checklist


- N/A

### Repository Checklist
